test/unknown_command.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/sh
+
+set -o errexit
+set -o nounset
+
+TMP="$(mktemp -d)"
+clean() { rm -rf "$TMP"; }
+trap clean EXIT
+
+"$1" unknown_command 1> "$TMP/stdout" 2> "$TMP/stderr" || exit_code=$?
```

Comment:
Also please see how we do it in other files for consistency. i.e. we use uppercase for variables: `EXIT_CODE`

Reply:
Also see the `&& EXIT_CODE="$?" || EXIT_CODE="$?"` pattern we use to avoid the `-z` cases you have below